tests/product/test_server_install.py

```diff
@@ -171,6 +171,45 @@ def assert_installed_with_regex_configs(self, master, slaves):
                                            '/etc/presto/config.properties',
                                            self.default_workers_config_regex_)
 
+    def test_install_with_java8_home(self, dummy=True):
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          "mv /usr/java/jdk1.8.0_40 /usr/")
+        topology = {"coordinator": "master",
+                    "workers": ["slave1", "slave2", "slave3"],
+                    "JAVA8_HOME": "/usr/jdk1.8.0_40/jre"}
+        self.upload_topology(topology)
+
+        cmd_output = self.installer.install(dummy)
+        expected = installed_all_hosts_output
+
+        actual = cmd_output.splitlines()
+        self.assertEqual(sorted(expected), sorted(actual))
+
+        for container in self.cluster.all_hosts():
+            self.installer.assert_installed(self, container)
+            self.assert_has_default_config(container)
+            self.assert_has_default_connector(container)
+
+    def test_install_failure_without_java8_home(self, dummy=True):
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          "mv /usr/java/jdk1.8.0_40 /usr/")
+        self.upload_topology()
+        cmd_output = self.installer.install(dummy)
+        actual = cmd_output.splitlines()
+        num_failures = 0
+        for line in enumerate(actual):
+            if str(line).find("Error: Required Java version"
+                              " could not be found") != -1:
+                num_failures += 1
```

Comment:
Honestly, we should be checking for the effects we care about as much as possible rather than looking at output all the time.

The check that the package isn't actually installed is the important check of the behavior of the system under the test conditions. I.e. that the package wasn't installed when Java can't be found.

The check on the output is a brittle check that we think we failed for the right reason. If the output or the behavior of the rpm changes for any reason, just checking the outputs can give us misleading results.